examples/3d/render_to_texture.rs

```diff
@@ -0,0 +1,216 @@
+use bevy::{
+    core_pipeline::{draw_3d_graph, node, AlphaMask3d, Opaque3d, Transparent3d},
+    prelude::*,
+    reflect::TypeUuid,
+    render::{
+        camera::{ActiveCameras, Camera, CameraProjection, ExtractedCameraNames, RenderTarget},
+        render_graph::{NodeRunError, RenderGraph, RenderGraphContext, SlotValue},
+        render_phase::RenderPhase,
+        render_resource::{
+            Extent3d, TextureDescriptor, TextureDimension, TextureFormat, TextureUsages,
+        },
+        renderer::RenderContext,
+        view::RenderLayers,
+        RenderApp, RenderStage,
+    },
+};
+
+pub const RENDER_IMAGE_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Image::TYPE_UUID, 13378939762009864029);
+
+pub const FIRST_PASS_DRIVER: &str = "first_pass_driver";
+pub const FIRST_PASS_CAMERA: &str = "first_pass_camera";
+
+#[derive(Component)]
+struct FirstPassCube;
+#[derive(Component)]
+struct MainPassCube;
+
+/// rotates the inner cube (first pass)
```

Comment:
From reading the code in a linear order, it's unclear what we expect to see rendered. Moving `main` to the start, and adding some expository comments will help users follow along.

Reply:
I've rearranged the code and added documentation to hopefully make the example more clear.